File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    for i in range(len(discount) - 9):
        check = [0] * len(want)
        cnt = sum(number)

        for j in range(10):
            if discount[i+j] in want:
                check[want.index(discount[i+j])] += 1
                cnt -=1
        if not cnt:
            if check == number:
                answer += 1
    
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
import java.util.*;

class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;
        Map<String, Integer> map = new HashMap<>();
        
        for(int i=0; i<want.length; i++) {
            map.put(want[i], number[i]);
        }
    
        for(int i=0; i<discount.length - 9; i++) {
            Map<String, Integer> tmp = new HashMap<>();
            
            for(int j=i; j<i+10; j++) {
                tmp.put(discount[j], tmp.getOrDefault(discount[j], 0)+1);
            }
            
            boolean flag = true;
            for(String m : map.keySet()) {
                if(map.get(m) != tmp.get(m)) {
                    flag = false;
                    break;
                }
            }
            
            if(flag) answer++;
        }
        
        return answer;
    }
}